core/src/main/java/com/scalar/db/common/ActiveTransactionManagedTransactionManager.java

```diff
@@ -48,7 +49,7 @@ public ActiveTransactionManagedTransactionManager(DatabaseConfig config) {
   private void add(DistributedTransaction transaction) throws TransactionException {
     if (activeTransactions.putIfAbsent(transaction.getId(), transaction) != null) {
       transaction.rollback();
-      throw new IllegalStateException(
+      throw new TransactionException(
```

Comment:
There is no change to expect or catch TransactionException in this PR. Does it mean we don't have any test case for this path? This might be a bit off-topic, though.

Reply:
Ah correct, we don't have tests for it. I will add them in this PR. Thanks!

Reply:
Added unit tests in 75bbd48d22c581069545ad6d7acb1787b99598bb.

@komamitsu Please take a look again when you have time!

Reply:
LGTM! Thank you!